datafusion/core/src/execution/context/mod.rs

```diff
@@ -1869,6 +1872,18 @@ impl SessionState {
     pub fn version(&self) -> &str {
         env!("CARGO_PKG_VERSION")
     }
+
+    /// Return extension context.
+    pub fn with_extension_context(&self) -> Option<Arc<dyn Any + Send + Sync>> {
```

Comment:
I think `with` prefixed functions are normally used to set fields in builder style code. I think accessors would be more normally named like this:

```suggestion
    pub fn extension_context(&self) -> Option<Arc<dyn Any + Send + Sync>> {
```